### PR TITLE
Fix an SVG focusing crash in IE11

### PR DESCRIPTION
**Pull request by ThaddeusJiang**
because #12763

### CODE REVIEW

**Comment by react-linter** [comment]
ReactDOM: size: 0.0%, gzip: 0.0%


  <details>
  <summary>Details of bundled changes.</summary>

  <p>Comparing: 9cf3733a9a43aa53d1a3cc94b2cd015b235b84cc...9871f37bd5550ec44c743db546c0ae968ad3e8a2</p>


  
## react-dom
File | Filesize Diff | Gzip Diff | Prev Size | Current Size | Prev Gzip | Current Gzip | ENV
 ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  --- 
react-dom.development.js | +0.1% | +0.1% | 626.32 KB | 626.88 KB | 145.82 KB | 145.96 KB | UMD_DEV
**react-dom.production.min.js** | **0.0%** | **0.0%** | **94.16 KB** | **94.18 KB** | **30.5 KB** | **30.51 KB** | **UMD_PROD**
react-dom.development.js | 0.0% | 0.0% | 610.68 KB | 610.75 KB | 141.81 KB | 141.83 KB | NODE_DEV
**react-dom.production.min.js** | **0.0%** | **:small_red_triangle:+0.1%** | **92.66 KB** | **92.69 KB** | **29.48 KB** | **29.5 KB** | **NODE_PROD**
ReactDOM-dev.js | 0.0% | 0.0% | 620.2 KB | 620.26 KB | 141.21 KB | 141.22 KB | FB_WWW_DEV
**ReactDOM-prod.js** | **0.0%** | **0.0%** | **269.19 KB** | **269.24 KB** | **50.57 KB** | **50.58 KB** | **FB_WWW_PROD**

  </details>
  
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: packages/react-dom/src/client/ReactInputSelection.js [hunk 1]
```diff
@@ -82,7 +83,7 @@ export function restoreSelection(priorSelectionInformation) {
       }
     }
 
-    priorFocusedElem.focus();
+    focusNode(priorFocusedElem);
```
Comment (gaearon) [inline packages/react-dom/src/client/ReactInputSelection.js]: Let’s just inline the try/catch here and leave a comment that this is for SVG in IE11.

**Comment by gaearon** [comment]
Would a fix like

```js
if (typeof priorFocusedElem.focus === 'function') {
  priorFocusedElem.focus();
}
```

be sufficient then?

**Comment by ThaddeusJiang** [comment]
Yes, It's better.

File: packages/react-dom/src/client/ReactInputSelection.js [hunk 2]
```diff
@@ -83,9 +83,9 @@ export function restoreSelection(priorSelectionInformation) {
       }
     }
 
-    try {
+    if (typeof priorFocusedElem.focus === 'function') {
       priorFocusedElem.focus();
-    } catch (e) {
+    } else {
       // The SVGElement don't have a focus method in IE11
       focusNode(priorFocusedElem);
```
Comment (gaearon) [inline packages/react-dom/src/client/ReactInputSelection.js]: Hmm, we don't need this part now, right?

**Comment by ThaddeusJiang** [comment]
@gaearon Yes, this fixed the IE11 issue.

**Comment by gaearon** [comment]
Thanks!

**Pull request merged**